src/site/topic25.rst

```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
 
 * `Here is a simple visualization of some popular sorting algorithms <https://www.toptal.com/developers/sorting-algorithms>`_
 
+
 Sorting Problem
 ===============
 
+* Given a collection of *things*
+    * Numbers
+    * Strings
+    * People
+    * Car Parts
+
+* The goal is to arrange the collection of *things* such that they are *in order*
+
+
+Order
+-----
+
+* What *in order* means however will depend on what is being sorted
+* For example
+    * Numbers come with an intrinsic ordering
+    * Strings could be ordered alphabetically
+        * Perhaps by length?
+        * Maybe based on the character's ascii values?
+        * Treat the strings as base 26 integers?
+    * Sort people based on height
+        * Or age?
+        * Or month and day of birthday?
+    * Car Parts by mass?
+        * Or production code?
+        * Date they were made?
+        * Volume?
+        * The amount of dust particles on the surface?
+
+* With numbers, we don't feel the need to ask
+* But with other things, we need to have a well defined ordering for the *things*
+    * The things themselves have the order-*ness* property, not a sorting algorithm
```

Comment:
May be worth bringing up [Comparable](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Comparable.html) (define natural ordering, and they've seen it a few times already) and [Comparator](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Comparator.html) (define an _unnatural_ ordering).

Reply:
mostly copy/paste